src/bun.js/bindings/NodeValidator.cpp

```diff
@@ -297,6 +335,30 @@ JSC_DEFINE_HOST_FUNCTION(jsFunction_validateArray, (JSC::JSGlobalObject * global
     auto value = callFrame->argument(0);
     auto name = callFrame->argument(1);
     auto minLength = callFrame->argument(2);
+    return V::validateArray(scope, globalObject, value, name, minLength);
+}
+JSC::EncodedJSValue V::validateArray(JSC::ThrowScope& scope, JSC::JSGlobalObject* globalObject, JSValue value, JSValue name, JSValue minLength)
+{
+    JSC::VM& vm = globalObject->vm();
+
+    if (minLength.isUndefined()) minLength = jsNumber(0);
+
+    if (!JSC::isArray(globalObject, value)) return Bun::ERR::INVALID_ARG_TYPE(scope, globalObject, name, "Array"_s, value);
```

Comment:
Instead of using `JSC::isArray`, and then otherwise treating it as a normal object, we can cast it to a `JSArray*` and then call `length` on that `JSArray`. It is stored as an unsigned integer.


```suggestion
   auto *array = jsDynamicCast<JSC::JSArray*>(value);
  if (!array)
  return Bun::ERR::INVALID_ARG_TYPE(scope, globalObject, name, "Array"_s, value);
```